fetcher.go

```diff
@@ -352,7 +354,15 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	err = cb(resource, &wrapped)
 	received = wrapped.len
 	if err != nil {
-		goLogger.Errorw("failed to read response", "err", err.Error())
+
+		_, ok := err.(ErrInvalidResponse)
```

Comment:
can you do this as [errors.As](https://pkg.go.dev/errors#example-As)?

Reply:
@willscott Done.